### PR TITLE
Fixing perf runs on helix

### PR DESCRIPTION
**Pull request by dsgouda**
- Archiving the entire Packages directory as payload since we do not have dependency lists in CoreCLR yet
- Revving xunit performance runner versions

@lt72 @MattGal @rahku

### CODE REVIEW

**Comment by dsgouda** [comment]
@dotnet-bot Test OSX x64 Checked Build and Test please

**Comment by dsgouda** [comment]
@dotnet-bot Test OSX x64 Checked Build and Test please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

File: tests/helixperftasks.targets [hunk 1]
```diff
@@ -1,4 +1,43 @@
-<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+    <UsingTask TaskName="RemoveDuplicateAssemblies" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll" >
```
Comment (MattGal) [inline tests/helixperftasks.targets]: I am very much not a fan of inline tasks.  Furthermore, there's already src\Microsoft.DotNet.Build.Tasks\RemoveDuplicatesWithLastOneWinsPolicy.cs  (in Microsoft.DotNet.Build.Tasks.dll).  If that one does not do what you want it'd be good to augment it til it did, or to add an entirely new task there to do this work.

File: tests/helixperftasks.targets [hunk 2]
```diff
@@ -1,4 +1,43 @@
...
+        <AssemblyPaths ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+        <UniqueAssemblyPaths ParameterType="Microsoft.Build.Framework.ITaskItem[]" Output="true" />
+      </ParameterGroup>
+      <Task>
+        <Using Namespace="System"/>
+        <Using Namespace="System.IO"/>
+        <Using Namespace="System.Collections.Generic"/>
+        <Code Type="Fragment" Language="cs">
+          <![CDATA[
+                  var UniqueAssemblyPathsList = new List<Microsoft.Build.Framework.ITaskItem>();
```
Comment (MattGal) [inline tests/helixperftasks.targets]: Inconsistent captalization.  These should all be camelCased, no?

File: tests/helixperftasks.targets [hunk 3]
```diff
@@ -33,6 +71,30 @@
 
   </Target>
 
+  <Target Name="RunTestsForProject">
+    <Message Text="This is a dummy target to appease DependsOnTargets for ArchiveTestBuild in publishtest.targets" Importance="Low" />
```
Comment (MattGal) [inline tests/helixperftasks.targets]: I think it makes more sense to change things there versus make a fake target, but not blocking for me at least.

File: tests/helixperftasks.targets [hunk 4]
```diff
@@ -57,18 +119,18 @@
                                TargetMonikers="@(TestTargetFramework)">
       <Output TaskParameter="ResolvedCopyLocalItems" ItemName="RequiredAssemblies" />
     </PrereleaseResolveNuGetPackageAssets>
+
+
     <ItemGroup>
       <AssemblyPaths Include="$([MSBuild]::MakeRelative($(PackagesDir), %(RequiredAssemblies.FullPath)))" />
+      <AssemblyFiles Include="@(RequiredAssemblies->'%(FileName)%(Extension)')"></AssemblyFiles>
+      <DupAssemblies Include="%(AssemblyFiles.Identity)" Condition="'@(AssemblyFiles->Count())' &gt; 1"> </DupAssemblies>
```
Comment (MattGal) [inline tests/helixperftasks.targets]: If you're going to shorten duplicate it's usually dupe, not dup.

File: tests/helixperftasks.targets (hunk 1 above)
Comment (dsgouda) [inline tests/helixperftasks.targets]: This particular task is doing something specific to CoreCLR only, would adding it to build tools be a good idea?

**Comment by dsgouda** [comment]
@dotnet-bot Test Windows_NT x64 Release Priority 1 Build and Test please

**Comment by MattGal** [comment]
If the CoreCLR folks are OK with the un-debug-ability of inline tasks, LGTM once you fix the casing.

**Comment by lt72** [comment]
+1 for Matt's concerns. 
I would suggest Deepak will open task to follow up with removing the inline task ASAP after release. In the interest of getting back up&running ASAP, I would suggest we proceed now with the current solution, which is not 'elegant', but it is correct.

**Comment by dsgouda** [comment]
I have opened a task to remove these inline tasks, we can probably piggyback on the changes made for functional tests under coreclr to run via helix.

**Comment by dsgouda** [comment]
@dotnet-bot Test Windows_NT x64 Release Priority 1 Build and Test please

**Comment by dsgouda** [comment]
@dotnet-bot Test Windows_NT x86 ryujit Checked Build and Test please

**Comment by dsgouda** [comment]
@dotnet-bot Test Windows_NT x86 legacy_backend Checked Build and Test please

**Pull request merged**